File: app/src/main/java/com/startogamu/zikobot/module/AppModule.java
package com.startogamu.zikobot.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    Application mApplication;

    /***
     * @param application
     */
    public AppModule(Application application) {
        mApplication = application;
    }

    @Provides
    @Singleton
    Application providesApplication() {
        return mApplication;
    }
}

File: app/src/main/java/dagger/AppModule.kt
package dagger

import com.startogamu.zikobot.ZikobotApp
import javax.inject.Singleton

/**
 * Created by blackbird-linux on 15/10/17.
 */
@Module class AppModule(val app: ZikobotApp) {
    @Provides @Singleton
    fun provideApp() = app
}